sdk/rust/src/entity_id.rs

```diff
@@ -53,6 +96,92 @@ impl EntityId {
     pub(crate) fn to_solidity_address(self) -> crate::Result<String> {
         EvmAddress::try_from(self).map(|it| it.to_string())
     }
+
+    pub(crate) fn generate_checksum(entity_id_string: &String, ledger_id: LedgerId) -> Checksum {
+        let p3 = 26 * 26 * 26 as usize; // 3 digits in base 26
+        let p5 = 26 * 26 * 26 * 26 * 26 as usize; // 5 digits in base 26
+        let m = 1_000_003 as usize; //min prime greater than a million. Used for the final permutation.
+        let w = 31 as usize; // Sum s of digit values weights them by powers of w. Should be coprime to p5.
+
+        let h = [ledger_id.to_bytes(), vec![0u8, 0u8, 0u8, 0u8, 0u8, 0u8]].concat();
```

Comment:
```suggestion
        let h = [ledger_id.to_bytes(), vec![0u8; 6]].concat();
```

I'd argue against creating a `Vec` here at all, but...